apollo-backend/src/main/java/io/logz/apollo/controllers/BlockerDefinitionController.java

```diff
@@ -136,6 +130,27 @@ public void addBlockerDefinition(String name, Integer environmentId, Integer ser
         assignJsonResponseToReq(req, HttpStatus.CREATED, blockerDefinition);
     }
 
+    private String initSingleRegionBlockerJsonConfig(Integer environmentId, Integer serviceId, Integer stackId, String availability, String blockerJsonConfiguration, Req req) {
+        String jsonConfig = null;
+
+        if (availability == null) {
+            logger.error("Could not initiate a SingleRegionBlocker without availability");
+            assignJsonResponseToReq(req, HttpStatus.BAD_REQUEST, "SingleRegionBlocker cannot be defined without specific availability");
+        } else if (environmentId != null || isStackEnvironmentType(stackId)) {
+            logger.error("Could not initiate a SingleRegionBlocker with environment or stack");
+            assignJsonResponseToReq(req, HttpStatus.BAD_REQUEST, "SingleRegionBlocker cannot be defined for a specific environment or stack");
+        } else if (blockerJsonConfiguration.equals("{}") || blockerJsonConfiguration == null) {
+            Optional<String> configOpt = blockerService.initSingleRegionBlockerConfiguration(serviceId, stackId);
+             if (configOpt.isPresent()) {
+                 jsonConfig = configOpt.get();
+             } else {
+                 logger.error("Could not initiate a SingleRegionBlocker without service or service-stack");
+                 assignJsonResponseToReq(req, HttpStatus.BAD_REQUEST, "SingleRegionBlocker cannot be defined without service(s)");
+             }
+        }
```

Comment:
please return an error in case !blockerJsonConfiguration.equals("{}") && blockerJsonConfiguration != null. It can't currently happen but might happen in the future.